ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -204,12 +206,63 @@ def resolve_service(self, provided_interface: str, service_class: str = "") -> S
 
 def _get_discovery_service_address() -> str:
     key_file_path = _get_key_file_path()
+    _ensure_discovery_service_started(key_file_path)
     _logger.debug("Discovery service key file path: %s", key_file_path)
     with _open_key_file(str(key_file_path)) as key_file:
         key_json = json.load(key_file)
         return "localhost:" + key_json["InsecurePort"]
 
 
+def _ensure_discovery_service_started(key_file_path: pathlib.Path):
+    """Make sure discovery service is started.
+
+    Check whether discovery service key file exists, if not start the discovery
+    service.
+
+    """
+    if _key_file_exists(key_file_path):
+        return
+
+    exe_file_path = _get_discovery_service_location()
+    _start_service(exe_file_path)
+
+
+def _get_discovery_service_location() -> pathlib.PurePath:
+    """Gets the location of the discovery service process executable.
+
+    Returns:
```

Comment:
docstring needed for private methods? if so we can add for all methods including params and return type as well.

Reply:
Posted in a separate PR: https://github.com/ni/measurementlink-python/pull/288#discussion_r1232811113

> Private helpers don't need full docstrings. The 1-line summary and any additional remarks are great, but the args/raises/returns sections are a bit much.

> ...we don't need to list the types in the docstring because Sphinx can read PEP 484 type annotations. Feel free to omit them in your new code.

Reply:
Updated docstring with 1-line summary